internal/storage/runtime.go

```diff
@@ -187,21 +188,31 @@ func (r *runtimeService) createContainerOrPodSandbox(systemContext *types.System
 
 	// Pull out a copy of the image's configuration.
 	// Ideally we would call imageID.imageRef(r.storageImageServer), but storageImageServer does not have access to private data.
-	ref, err := istorage.Transport.NewStoreReference(r.storageImageServer.GetStore(), nil, template.imageID.privateID)
-	if err != nil {
-		return ContainerInfo{}, err
-	}
-	image, err := ref.NewImage(r.ctx, systemContext)
-	if err != nil {
-		return ContainerInfo{}, err
-	}
-	defer image.Close()
+	var imageConfig *v1.Image
+	if isRuntimePullImage {
```

Comment:
I don’t know why this parameter needs to exist, and I’m generally suspicious of call trees where a function deep down behaves _significantly_ differently based on an option (and I do consider reading local files vs. reaching out to a network significant); can’t the code be refactored to
- read `imageConfig` earlier
- possibly turn the reading of `imageConfig` into a stand-alone helper,
- then move the read of `imageConfig` to the callers?

Callers could then pass the `imageConfig` directly instead of the `isRuntimePullImage` flag. That’s not any more complex, at least.